src/shared/copy.c

```diff
@@ -1000,9 +1005,9 @@ static int fd_copy_directory(
                                 return r;
                 }
 
-                if (set_contains(denylist, &buf)) {
-                        log_debug("%s/%s is in the denylist, skipping", from, de->d_name);
-                        continue;
+                if (hashmap_contains(denylist, &buf) && PTR_TO_INT(hashmap_get(denylist, &buf)) == DENY_INODE) {
```

Comment:
the hashmap_contains() call is unnecessary if you add DENY_DONT as suggested below